README.md
# flutter_text_form_field

A new Flutter package that provide already made text form field

##Usage

[Example] (https://github.com/Destiny-Ed/flutter_text_form_field/blob/main/example/example_app.dart)

To use this package :
*add the dependency to your [pubspec.yaml] file

```yaml
    dependencies:
        flutter:
            sdk : flutter
        flutter_text_form_field: ^0.0.3
```

```dart
    //...
    Example Usage
    //...

import 'package:flutter/material.dart';
import 'package:flutter_text_form_field/flutter_text_form_field.dart';

class Login extends StatefulWidget {
  @override
  _LoginState createState() => _LoginState();
}

class _LoginState extends State<Login> {
  final TextEditingController _emailController = TextEditingController();
  final TextEditingController _passwordController = TextEditingController();

  @override
  Widget build(BuildContext context) {
    return Scaffold(
      body: SingleChildScrollView(
        child: Container(
          height: MediaQuery.of(context).size.height,
          width: MediaQuery.of(context).size.width,
          decoration: BoxDecoration(color: Colors.purple),
          child: Column(
            children: [
              const SizedBox(
                height: 50,
              ),
              Text(
                "Continue to your account",
                style: TextStyle(color: Colors.white, fontSize: 17),
              ),
              const SizedBox(
                height: 100,
              ),
              Expanded(
                child: Container(
                    padding: const EdgeInsets.all(30.0),
                    width: MediaQuery.of(context).size.width,
                    height: MediaQuery.of(context).size.height,
                    decoration: BoxDecoration(
                      color: Colors.white,
                      borderRadius: const BorderRadius.only(
                          topLeft: Radius.circular(20),
                          topRight: Radius.circular(20)),
                    ),
                    child: Column(
                      mainAxisAlignment: MainAxisAlignment.spaceEvenly,
                      children: [
                        Column(
                          children: [
                            CustomTextField(
                              _emailController,
                              hint: 'Email',
                              password: false,
                            ),
                            const SizedBox(
                              height: 20,
                            ),
                            CustomTextField(
                              _passwordController,
                              hint: 'Password',
                              obscure: true,
                            ),
                            const SizedBox(
                              height: 8,
                            ),
                            GestureDetector(
                              onTap: () {
                                //Forgot Password Page
                              },
                              child: Container(
                                  alignment: Alignment.centerRight,
                                  child:
                                      const Text("Can't remember password?")),
                            ),
                            const SizedBox(
                              height: 20,
                            ),
                          ],
                        ),
                        GestureDetector(
                          onTap: () {
                            print(_emailController.text);
                            print(_passwordController.text);
                            //Validate User Inputs
                          },
                          child: Container(
                            padding: const EdgeInsets.all(15.0),
                            width: MediaQuery.of(context).size.width,
                            decoration: BoxDecoration(
                              color: Colors.purple,
                              borderRadius: BorderRadius.circular(10),
                            ),
                            child: Text(
                              "Login",
                              style: TextStyle(color: Colors.white),
                              textAlign: TextAlign.center,
                            ),
                          ),
                        ),
                        GestureDetector(
                          onTap: () {
                            //Navigate to Register Page
                          },
                          child: Text(
                            "Want to join?",
                            style: TextStyle(fontSize: 17),
                          ),
                        ),
                      ],
                    )),
              ),
            ],
          ),
        ),
      ),
    );
  }
}

```


## Getting Started

This project is a starting point for a Dart
[package](https://flutter.dev/developing-packages/),
a library module containing code that can be shared easily across
multiple Flutter or Dart projects.

For help getting started with Flutter, view our 
[online documentation](https://flutter.dev/docs), which offers tutorials, 
samples, guidance on mobile development, and a full API reference.


CHANGELOG.md
## [0.0.1] - TODO: Add release date.

* TODO: An easy to use flutter custom text form field package written in dart

## [0.0.2] - Febuary 14 2021

* Added contructor for changing background color

## [0.0.3] - August 13 2021

* Migrated to null safety


lib/flutter_text_form_field.dart
library flutter_text_form_field;

/// A Flutter Custom Text Form FieldSetElement
///
import 'package:flutter/material.dart';

class CustomTextField extends StatefulWidget {
  final TextEditingController _controller;
  bool obscure;
  Color? backgroundColor;
  Color? pwdVisibilityColor;
  String hint;
  bool password;
  BoxBorder? border;
  BorderRadiusGeometry? borderRadius;
  bool readOnly;
  Function(String)? onPress;
  Function()? onEditingComplete;
  Function(String)? onFieldSubmitted;
  String Function(String?)? validator;

  CustomTextField(this._controller,
      {Key? key,
      this.password = true,
      this.obscure = false,
      this.hint = '',
      this.border,
      this.pwdVisibilityColor,
      this.borderRadius,
      this.readOnly = false,
      this.backgroundColor,
      this.onPress,
      this.onEditingComplete,
      this.onFieldSubmitted,
      this.validator})
      : super(key: key);

  @override
  _CustomTextFieldState createState() => _CustomTextFieldState();
}

class _CustomTextFieldState extends State<CustomTextField> {
  Icon? visibility;

  @override
  Widget build(BuildContext context) {
    return Container(
      padding: const EdgeInsets.only(left: 10.0),
      decoration: BoxDecoration(
        color: widget.backgroundColor == null
            ? Colors.white
            : widget.backgroundColor,
        borderRadius: widget.borderRadius ?? BorderRadius.circular(10),
        border: widget.border ?? Border.all(color: Colors.purple, width: 2.0),
      ),
      child: TextFormField(
        key: widget.key,
        controller: widget._controller,
        obscureText: widget.obscure,
        decoration: InputDecoration(
          border: InputBorder.none,
          suffixIcon: widget.password == true
              ? GestureDetector(
                  key: widget.key,
                  onTap: () {
                    //Show and hide password
                    if (widget.obscure == true) {
                      setState(() {
                        widget.obscure = false;
                        visibility = Icon(Icons.visibility_off,
                            color: widget.pwdVisibilityColor == null
                                ? Colors.black
                                : widget.pwdVisibilityColor);
                      });
                    } else {
                      setState(() {
                        widget.obscure = true;
                        visibility = Icon(Icons.visibility,
                            color: widget.pwdVisibilityColor == null
                                ? Colors.black
                                : widget.pwdVisibilityColor);
                      });
                    }
                  },
                  child: visibility == null
                      ? Icon(Icons.visibility,
                          color: widget.pwdVisibilityColor == null
                              ? Colors.black
                              : widget.pwdVisibilityColor)
                      : visibility,
                )
              : const Text(""),
          isDense: true,
          contentPadding: const EdgeInsets.only(
            top: 15,
            bottom: 10,
          ),
          hintText: widget.hint,
        ),
        readOnly: widget.readOnly,
        onChanged: widget.onPress,
        onEditingComplete: widget.onEditingComplete,
        onFieldSubmitted: widget.onFieldSubmitted,
        validator: widget.validator,
      ),
    );
  }
}


example/example_app.dart
import 'package:flutter/material.dart';
import 'package:flutter_text_form_field/flutter_text_form_field.dart';

class Login extends StatefulWidget {
  @override
  _LoginState createState() => _LoginState();
}

class _LoginState extends State<Login> {
  final TextEditingController _emailController = TextEditingController();
  final TextEditingController _passwordController = TextEditingController();

  @override
  Widget build(BuildContext context) {
    return Scaffold(
      body: SingleChildScrollView(
        child: Container(
          height: MediaQuery.of(context).size.height,
          width: MediaQuery.of(context).size.width,
          decoration: BoxDecoration(color: Colors.purple),
          child: Column(
            children: [
              const SizedBox(
                height: 50,
              ),
              Text(
                "Continue to your account",
                style: TextStyle(color: Colors.white, fontSize: 17),
              ),
              const SizedBox(
                height: 100,
              ),
              Expanded(
                child: Container(
                    padding: const EdgeInsets.all(30.0),
                    width: MediaQuery.of(context).size.width,
                    height: MediaQuery.of(context).size.height,
                    decoration: BoxDecoration(
                      color: Colors.white,
                      borderRadius: const BorderRadius.only(
                          topLeft: Radius.circular(20),
                          topRight: Radius.circular(20)),
                    ),
                    child: Column(
                      mainAxisAlignment: MainAxisAlignment.spaceEvenly,
                      children: [
                        Column(
                          children: [
                            CustomTextField(
                              _emailController,
                              hint: 'Email',
                              password: false,
                            ),
                            const SizedBox(
                              height: 20,
                            ),
                            CustomTextField(
                              _passwordController,
                              hint: 'Password',
                              obscure: true,
                            ),
                            const SizedBox(
                              height: 8,
                            ),
                            GestureDetector(
                              onTap: () {
                                //Forgot Password Page
                              },
                              child: Container(
                                  alignment: Alignment.centerRight,
                                  child:
                                      const Text("Can't remember password?")),
                            ),
                            const SizedBox(
                              height: 20,
                            ),
                          ],
                        ),
                        GestureDetector(
                          onTap: () {
                            print(_emailController.text);
                            print(_passwordController.text);
                            //Validate User Inputs
                          },
                          child: Container(
                            padding: const EdgeInsets.all(15.0),
                            width: MediaQuery.of(context).size.width,
                            decoration: BoxDecoration(
                              color: Colors.purple,
                              borderRadius: BorderRadius.circular(10),
                            ),
                            child: Text(
                              "Login",
                              style: TextStyle(color: Colors.white),
                              textAlign: TextAlign.center,
                            ),
                          ),
                        ),
                        GestureDetector(
                          onTap: () {
                            //Navigate to Register Page
                          },
                          child: Text(
                            "Want to join?",
                            style: TextStyle(fontSize: 17),
                          ),
                        ),
                      ],
                    )),
              ),
            ],
          ),
        ),
      ),
    );
  }
}
